### DISCUSSION TITLE
Discord Bot was working when i made it now is not working?

### DISCUSSION THREAD

**Discussion by rokipet** [Rapptz/discord.py#9659] [General] [answered]
well i made this discord bot yesterday it was working like charm now today i got something that the button was give me errors
this is the error
raceback (most recent call last):
  File "C:\Users\Yosbel.Gonzalez001\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\ui\view.py", line 427, in _scheduled_task
    await item.callback(interaction)
  File "C:\Users\Yosbel.Gonzalez001\Downloads\Bots\Registracion Discord Bot\main.py", line 42, in registration_button
    user = interaction.user
           ^^^^^^^^^^^^^^^^
AttributeError: 'Button' object has no attribute 'user'

 tried to do some changes i went to ubuntu and the codes work but on windows didnt work any more

this is the part of the code that may need attention
 @discord.ui.button(label="Registración", style=discord.ButtonStyle.green)
    async def registration_button(self, button: Button, interaction: discord.Interaction):
        user = interaction.user
        if user.id in pending_registrations:
            await user.send("Ya estás en proceso de registro.")
        else:
            pending_registrations[user.id] = {"step": 1}
            await user.send("Por favor, introduce tu correo electrónico:")

**Comment by itswilliboy** [comment, marked as answer]
Swap the `button` and the `interaction`arguments. So it becomes `(self, interaction, button):`.

You probably have different versions of the library on Windows and Ubuntu.

**Reply by rokipet** [reply]
Wow thanks very much 5 days trying to figure out lol im a starter

**Reply by itswilliboy** [reply]
No worries, have fun!
